custom_modules/gateway/gateway.js

```diff
@@ -0,0 +1,42 @@
+class NetworkGateway {
+  constructor() {
+    this.endpointURL = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games';
+    this.id = '';
+  }
+
+  getID = async (key) => {
+    await fetch(`${this.endpointURL}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ name: key }),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        const stringResult = data.result;
+        this.id = stringResult.substring(13, 34);
+      })
+      .catch((error) => {
+        throw error;
+      });
+  }
+
+  insertData = async (_name, _score) => {
+    const data = await fetch(`${this.endpointURL}/${this.id}/scores`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ user: _name, score: _score }),
+    });
+    return data.json();
+  }
+
+  fetchData = async () => {
+    const data = await fetch(`${this.endpointURL}/${this.id}/scores`);
+    return data.json();
+  }
+}
```

Comment:
- [ ] There seems to be no data retrieved from this request, after refreshing the page, the scores are completely removed from the board.
![Screenshot from 2022-03-02 01-11-31](https://user-images.githubusercontent.com/33767951/156264769-9683c11f-0543-4057-87b5-c556e797154a.png)

Reply:
please could you confirm this again? it's not the same on my side. you could check your network or something

Reply:
@harlexkhal, sorry for the confusion, I didn't mention anything about localStorage, however, data submitted to the API should be retrieved and rendered when clicking `Refresh` or just by refreshing the page. If you noticed in the network tab the retrieved array is empty.